### PR TITLE
[TAO-7459] Extract release notes during the release of an extension 

### PR DESCRIPTION
**Pull request by AntonTsymuk**
- Refactor release functionality into separate module in purpose to be able to test it
- Partially migrate application to use github v4 api
- Fix issue with extraction of release notes which was caused by github api search query restriction of 256 characters
- Introduce command line arguments to increase flexibility on release configuration
- Cover changes with unit tests

### CODE REVIEW

File: tests/unit/release/confirmRelease/test.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    prompt: () => ({ extension, taoRoot }),
+};
+const version = '1.1.1';
+const taoInstance = {
+    getExtensions: () => [],
+    isInstalled: () => true,
+    isRoot: () => ({ root: true, dir: taoRoot }),
+    parseManifest: () => ({ version, name: extension })
+};
+const taoINstanceFactory = sandbox.stub().callsFake(() => taoInstance);
```
Comment (krampstudio) [inline tests/unit/release/confirmRelease/test.js]: is the case ok `taoINstanceFactory` intentional ?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@oat-sa/tao-extension-release",
-  "version": "0.3.1",
+  "version": "0.3.2",
```
Comment (krampstudio) [inline package.json]: please upgrade to `0.4.0`, release note support is a new feature IMO

File: tests/unit/release/compileAssets/test.js [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+};
+const version = '1.1.1';
+const taoInstance = {
+    buildAssets: () => { },
+    getExtensions: () => [],
+    isInstalled: () => true,
+    isRoot: () => ({ root: true, dir: taoRoot }),
+    parseManifest: () => ({ version, name: extension })
+};
+const taoINstanceFactory = sandbox.stub().callsFake(() => taoInstance);
```
Comment (krampstudio) [inline tests/unit/release/compileAssets/test.js]: same case issue

File: tests/unit/release/compileAssets/test.js [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+    await release.selectTaoInstance();
+    await release.selectExtension();
+    await release.verifyBranches();
+
+    sandbox.stub(taoInstance, 'buildAssets');
+
+    await release.compileAssets();
+
+    t.equal(taoInstance.buildAssets.callCount, 1, 'Assets has been builded');
```
Comment (krampstudio) [inline tests/unit/release/compileAssets/test.js]: `built`

**Comment by AntonTsymuk** [comment]
@krampstudio PR has been updated according to you comments. Please check

**Pull request merged**